cilium-health/launch/endpoint.go

```diff
@@ -294,7 +295,7 @@ func LaunchAsEndpoint(owner endpoint.Owner, n *node.Node, mtuConfig mtu.Configur
 		return nil, err
 	}
 	if !ep.SetStateLocked(endpoint.StateWaitingToRegenerate, "initial build of health endpoint") {
-		endpointmanager.Remove(ep)
+		endpointmanager.Remove(ep, &sync.WaitGroup{})
```

Comment:
This should be nil - alternatively create two separate 'Remove' functions (one that asynchronously releases endpoint ID, one that doesn't).